GCEWindowsAgent/addresses.go

```diff
@@ -17,32 +17,77 @@ package main
 import (
 	"net"
 	"reflect"
+	"strconv"
 	"strings"
 
 	"github.com/GoogleCloudPlatform/compute-image-windows/logger"
+	"github.com/go-ini/ini"
 )
 
-var addressDisabled = false
-var addressKey = regKeyBase + `\ForwardedIps`
+var (
+	addressDisabled  = false
+	addressKey       = regKeyBase + `\ForwardedIps`
+	oldWSFCAddresses string
+	oldWSFCEnable    bool
+)
 
 type addresses struct {
 	newMetadata, oldMetadata *metadataJSON
+	config                   *ini.File
 }
 
 func (a *addresses) diff() bool {
-	return !reflect.DeepEqual(a.newMetadata.Instance.NetworkInterfaces, a.oldMetadata.Instance.NetworkInterfaces) ||
-		a.newMetadata.Instance.Attributes.WSFCAddresses != a.oldMetadata.Instance.Attributes.WSFCAddresses ||
-		a.newMetadata.Instance.Attributes.EnableWSFC != a.oldMetadata.Instance.Attributes.EnableWSFC
-}
+	wsfcAddresses := a.config.Section("wsfc").Key("addresses").String()
+	if len(wsfcAddresses) == 0 {
```

Comment:
For consistency and ease of understanding the ordering, you should handle the case where the config file is set before handling metadata cases. Also, if a user has set the wsfcAddresses variable (and left it blank) should that be treated as an override to prevent metadata from being used?

FWIW, I don't think the logic here is very complicated, but I find the code fairly confusing as written (specific to this function).